src/components/Layout.tsx

```diff
@@ -70,11 +78,67 @@ interface NavBarProps {
   endpoint: GraphQLEndPoint;
 }
 export const NavBar: React.FC<NavBarProps> = ({ className, endpoint }) => {
-  const onSearch = (value: string) => {
+  const client = useApolloClient();
+  const onSearch = async (value: string) => {
+    value = value.trim();
     if (value.match(/^[0-9a-fA-F]{64}$/)) {
-      navigate(`/search/?${value}`);
+      try {
+        const data = await client.query<BlockByHashQuery>({
+          query: BlockByHashDocument,
+          variables: {hash: value},
+        });
+        if (
+          data.data.chainQuery.blockQuery
+          && data.data.chainQuery.blockQuery.block
+          && data.data.chainQuery.blockQuery.block.hash
```

Comment:
```suggestion
          data.data.chainQuery.blockQuery?.block?.hash
```

Prefer optional chaining if available. Also applies to other usages as well.

Reply:
I tried this, and get something like `optional chaining isn't currently enabled`. Do I have to update something?

Reply:
I think it might be better off having this as a separate PR, since there are already 2 more PRs that would be impacted by this change (#162, #170). What do you think?